src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/CompiledPageRouteModelProvider.cs

```diff
@@ -149,5 +152,23 @@ private void CreateModels(PageRouteModelProviderContext context)
                 }
             }
         }
+
+        internal static string GetRouteTemplate(CompiledViewDescriptor viewDescriptor)
+        {
+            if (viewDescriptor.ViewAttribute != null)
+            {
+                return ((RazorPageAttribute)viewDescriptor.ViewAttribute).RouteTemplate;
```

Comment:
Hmm isnt it possible for an extender to have a non RazorPageAttribute here? If not, then awesome, if so then i'd recommend `as` casting to see if it is of the right type.

Reply:
Not really. The attribute is guaranteed to either be `null` or an instance of `RazorPageAttribute` (since that's what the discovery of this looks like).